lib/guard.js

```diff
@@ -131,18 +134,29 @@ const MetricsGuard = class MetricsGuard extends stream.Transform {
 			this.registry.set(metric.name, entry);
 		}
 
+		let invalidLabels = [];
 		// push all label permutations on the metric into the
 		// Set for the metric in the registry. the size of
 		// this Set is compared with the threshold
 		metric.labels.forEach((label) => {
-			entry.add(`${label.name}${SEPARATOR}${label.value}`);
+			if (label && (label.name === null || typeof label.name === "undefined")) {
```

Comment:
Should we just scan all object properties for null / undefined values instead of targeting specific properties?

Reply:
sounds like a good idea 👍

Reply:
I was looking at the types and there `null` is valid for values, so I guess we should only look for `undefined` then.

Reply:
I added a loop on all properties, let me know what you think.

Reply:
I looked at this some more, `metric.labels` is an array of `MetricLabel`s and that means I shouldn't have to look for anything but the `name` and `value` should as @wkillerud said be allowed to be `null / undefined`